fe/src/main/java/org/apache/doris/analysis/SelectStmt.java

```diff
@@ -341,7 +345,11 @@ public void analyze(Analyzer analyzer) throws AnalysisException, UserException {
                 // of expr child and depth limits (toColumn() label may call toSql()).
                 item.getExpr().analyze(analyzer);
                 if (item.getExpr().contains(Predicates.instanceOf(Subquery.class))) {
-                    throw new AnalysisException("Subquery is not supported in the select list.");
+                    if (item.getExpr() instanceof CaseExpr) {
```

Comment:
How about the case expr in the where predicate?

Reply:
this is not allowed by in predicate or other predicate